src/App.tsx

```diff
@@ -1,61 +1,103 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './react-app-env';
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosFromServer);
+  const [title, setTitle] = useState('');
+  const [selectUser, setSelectUser] = useState(0);
+  const [submit, setSubmit] = useState(false);
+
+  const largestId = Math.max(...todos.map(todo => todo.id));
+
+  const addTodo = () => {
+    const newTodo: Todo = {
+      id: largestId + 1,
+      title,
+      completed: false,
+      userId: +selectUser,
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleSubmit = (e: React.MouseEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    setSubmit(true);
+
+    if (title && selectUser) {
+      addTodo();
+      setSubmit(false);
+      setTitle('');
+      setSelectUser(0);
+    }
```

Comment:
1. Don't call change of same state twice in one function call.
2. Revert logic - if something wrong - return. In other case proceed with adding new todo.
3. You need to check if title is string totally with spaces (blank string, not empty)